src/com/ms/silverking/log/AsyncLogger.java

```diff
@@ -1,32 +1,31 @@
 /**
- * 
+ *
  */
 package com.ms.silverking.log;
 
 import java.util.concurrent.BlockingQueue;
 
 import com.ms.silverking.thread.ThreadUtil;
 
-
 class AsyncLogger implements Runnable {
-    private final BlockingQueue<LogEntry>    logQueue;
-    
-    AsyncLogger(BlockingQueue<LogEntry> logQueue) {
-        this.logQueue = logQueue;
-        ThreadUtil.newDaemonThread(this, "AsyncLogger").start();
-    }
-    
-    public void run() {
-        // FUTURE - we can't really shut this one down
-        // change implementation if we ever want to do that
-        while (true) {
-            LogEntry    entry;
-            
-            try {
-                entry = logQueue.take();
-                entry.log();
-            } catch (InterruptedException ie) {
-            }
-        }
+  private final BlockingQueue<LogEntry> logQueue;
+
+  AsyncLogger(BlockingQueue<LogEntry> logQueue) {
+    this.logQueue = logQueue;
+    ThreadUtil.newDaemonThread(this, "AsyncLogger").start();
+  }
+
+  public void run() {
+    // FUTURE - we can't really shut this one down
+    // change implementation if we ever want to do that
+    while (true) {
+      LogEntry entry;
+
+      try {
+        entry = logQueue.take();
+        entry.log();
+      } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)